src/main/proto/com/kodypay/grpc/ecom/v1/ecom.proto

```diff
@@ -62,28 +62,61 @@ message PaymentDetailsRequest {
 
 message PaymentDetailsResponse {
   oneof result {
-    Response response = 1;
+    PaymentDetails response = 1;
```

Comment:
I personally prefer to have a generic naming for every service request to have a top level structure that is always a tuple with response/error, similar to the structured restful where they return "data", "links", "error", etc.
I would put the PaymentDetails inside the Response:
```suggestion
    Response response = 1;
    
    message Response {
       PaymentDetails paymentDetails = 1;
       ...
     }
```
I also believe that there will be occasions where a PaymentDetails from a initial payment request might not be the same as getting payment details. I personally prefer to use less logic with optionals, and more obvious strict data structures.

Reply:
The logic behind using PaymentDetails message is because now it's a shared message between the GetDetails API and the GetPayments API given the message is exactly the same.
My idea is that if they start to drift away then we need to introduce a new message and deprecate the old one (possibly for the GetPayments API given we maybe want less details here)
I don't envision to use optionals to map two inherently different objects. Here optionals are used to mark fields that could maybe not be set (like the paid_date)

Reply:
Designing for future or permanent backwards compatibility needs to have this compromise of duplicating data when there is a possibility of drifting away. We don't want "ever" to use deprecation as part of a natural evolution of a public facing API. Planning for future backwards compatibility is adamant for a successful API.

Reply:
I see the point. There's no guarantee, though, that we are not going to deprecate fields but is better than deprecating fields containing messages